fuel-core/src/database.rs

```diff
@@ -332,6 +335,20 @@ impl InterpreterStorage for Database {
 
 impl TxPoolDb for Database {}
 
+impl BlockProducerDatabase for Database {
+    fn get_block(
+        &self,
+        fuel_height: BlockHeight,
+    ) -> anyhow::Result<Option<Cow<FuelBlockDb>>> {
+        // let v = self.get(fuel_height.into(), Column::FuelBlocks);
+        // let v = v?;
+
+        let id = self.block_hash(fuel_height.into())?;
+        let block = self.storage::<FuelBlocks>().get(&id)?.unwrap_or_default();
```

Comment:
Since the return type is Option<Cow<Block>>, we should return None instead of using `unwrap_or_default()`